vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -183,7 +181,23 @@ type internal FSharpProjectOptionsManager
     /// Get the options for a project
     member this.TryGetOptionsForProject(projectId: ProjectId) = 
         match projectTable.TryGetValue(projectId) with
-        | true, ((_referencedProjects, options), _) -> Some options 
```

Comment:
I'm concerned by this change.  the whole point of ``TryGetOptionsForProject`` and ``projectTable`` is to maintain the canonical ``FSharpProjectOptions`` object for the project so it doesn't need to be recomputed - and FCS can use ``Stamp`` to tell if the project options of dependent projects have changed.  For example, each time you make a new options object you should bump ``Stamp``.  

But with this change, the options stored in ``projectTable`` are no longer canonical and up-to-date - instead you have to go through this routine to get the latest options. 

It would be better to update options entry in ``projectTable`` either here (on-demand) or at some earlier point when we know that the ``CommandLineOptions``/``sourcePaths``/``otherOptions`` results change

Reply:
Note that failing to keep the ``projectTable`` canonical may have correctness of performance effects on incorrect or recomputation of project options.